modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/web/internal/info/item/provider/test/ObjectEntryInfoItemFieldValuesProviderTest.java

```diff
@@ -0,0 +1,214 @@
+	@ClassRule
+	@Rule
+	public static final AggregateTestRule aggregateTestRule =
+		new LiferayIntegrationTestRule();
+
+	@Before
+	public void setUp() throws Exception {
+		PropsUtil.addProperties(
+			UnicodePropertiesBuilder.setProperty(
+				"feature.flag.LPS-176083", "true"
+			).build());
+
+		_group = GroupTestUtil.addGroup();
+
+		_businessTypeAttachmentObjectField =
+			_getBusinessTypeAttachmentObjectField();
+
+		_objectDefinition =
+			_objectDefinitionLocalService.addCustomObjectDefinition(
+				TestPropsValues.getUserId(), false,
+				LocalizedMapUtil.getLocalizedMap(RandomTestUtil.randomString()),
+				"A" + RandomTestUtil.randomString(), null, null,
+				LocalizedMapUtil.getLocalizedMap(RandomTestUtil.randomString()),
+				ObjectDefinitionConstants.SCOPE_SITE,
+				ObjectDefinitionConstants.STORAGE_TYPE_DEFAULT,
+				Arrays.asList(_businessTypeAttachmentObjectField));
+
+		_objectDefinition =
+			_objectDefinitionLocalService.publishCustomObjectDefinition(
+				TestPropsValues.getUserId(),
+				_objectDefinition.getObjectDefinitionId());
+	}
+
+	@After
+	public void tearDown() {
+		PropsUtil.addProperties(
+			UnicodePropertiesBuilder.setProperty(
+				"feature.flag.LPS-176083", "false"
+			).build());
+	}
+
+	@Test
+	public void testObjectEntryInfoItemFieldValuesProviderWithAttachment()
+		throws Exception {
+
+		FileEntry fileEntry = TempFileEntryUtil.addTempFileEntry(
+			TestPropsValues.getGroupId(), TestPropsValues.getUserId(),
+			_objectDefinition.getPortletId(),
+			TempFileEntryUtil.getTempFileName("document.txt"),
+			FileUtil.createTempFile(RandomTestUtil.randomBytes()),
+			ContentTypes.TEXT_PLAIN);
+
+		ObjectEntry objectEntry = _objectEntryLocalService.addObjectEntry(
+			TestPropsValues.getUserId(), _group.getGroupId(),
+			_objectDefinition.getObjectDefinitionId(),
+			HashMapBuilder.<String, Serializable>put(
+				_businessTypeAttachmentObjectField.getName(),
+				fileEntry.getFileEntryId()
+			).build(),
+			ServiceContextTestUtil.getServiceContext());
+
+		InfoItemFieldValuesProvider<ObjectEntry> infoItemFieldValuesProvider =
+			_infoItemServiceRegistry.getFirstInfoItemService(
+				InfoItemFieldValuesProvider.class,
+				_objectDefinition.getClassName());
+
+		InfoItemFieldValues infoItemFieldValues =
+			infoItemFieldValuesProvider.getInfoItemFieldValues(objectEntry);
+
+		Assert.assertNotNull(infoItemFieldValues);
+
+		ObjectField objectField = _objectFieldLocalService.fetchObjectField(
+			_objectDefinition.getObjectDefinitionId(),
+			_businessTypeAttachmentObjectField.getName());
+
+		Assert.assertNotNull(objectField);
+
+		for (String name :
+				new String[] {
+					_businessTypeAttachmentObjectField.getName(),
+					objectField.getObjectFieldId() + "#downloadURL",
+					objectField.getObjectFieldId() + "#fileName",
+					objectField.getObjectFieldId() + "#mimeType",
+					objectField.getObjectFieldId() + "#size"
+				}) {
+
+			Assert.assertNotNull(infoItemFieldValues.getInfoFieldValue(name));
+		}
+	}
```

Comment:
@jkappler 

Is it difficult to cover the exact amount for [each case](https://github.com/liferay-objects/liferay-portal/pull/1852/files#diff-c1453c1728782646fb2f30ef802cb83b76165451db8e8e1d1ed70415e9c492d1R146-R151)?

Reply:
hi @MarinhoFeliphe, for downloadURL a bit since it has a timestamp

Reply:
@MarinhoFeliphe, I will update the PR on Monday. Thanks again for all the help!